augur/parse.py

```diff
@@ -162,11 +163,12 @@ def run(args):
     # field to index the dictionary and the data frame
     meta_data = {}
 
-    if 'name' in args.fields:
-        strain_key = 'name'
-    elif 'strain' in args.fields:
-        strain_key = 'strain'
-    else:
+    strain_key = None
+    for possible_id in [args.output_id_field, DEFAULT_ID_COLUMNS]:
```

Comment:
I like @victorlin's [suggested alternative to this block](https://github.com/nextstrain/augur/pull/1403#discussion_r1471787967), but I wanted to mention that this code on line 167 doesn't do what you want it to do, @j23414. Instead of looping through a list of the user-provided output id field and the default ids, this will loop through the user-provided field and the tuple of defaults:

```python
>>> DEFAULT_ID_COLUMNS = ("strain", "name")
>>> output_id_field = "accession"
>>> for possible_id in [output_id_field, DEFAULT_ID_COLUMNS]:
...    print(possible_id)
...
accession
('strain', 'name')
```

You could make this work as expected a couple of ways including [iterable unpacking with the `*` operator](https://docs.python.org/3/reference/expressions.html#expression-lists) or list concatenation.

```python
# Iterable unpacking with *
>>> for possible_id in [output_id_field, *DEFAULT_ID_COLUMNS]:
...    print(possible_id)
...
accession
strain
name

# List concatenation
>>> for possible_id in [output_id_field] + list(DEFAULT_ID_COLUMNS):
...    print(possible_id)
...
accession
strain
name
```

Reply:
gah, thanks! 😅